devolving_music/lib/song_scores.py

```diff
@@ -12,165 +13,174 @@
+        return random.choice(key_list)
 
-    @staticmethod
-    def get_info_sort(song_submissions_scored: Iterable["SongSubmission"]):
+    def get_compare_submission_linear(self, submission_key):
+        compare_list = self.comparison_submissions
+        if(len(compare_list)==0):
+            return self.get_compare_submission_random(submission_key)
+        first_sub_id=compare_list[-1].first_submission.id
+        if(submission_key==first_sub_id):
+            return self.get_compare_submission_random(submission_key)
+        return first_sub_id
 
-        info_submissions = []
+        # once you have a critical number of comparisons then pull from quality list
+        return random.choice(key_list)
 
-        # sort by info_submissions=song_submissions_scored.info
-        return info_submissions
 
-    @staticmethod
-    def get_energy_sort(song_submissions_scored: Iterable["SongSubmission"]):
-
-        # song_submissions_scored is a list of song submissions
-
-        # that have been scored with current comparisons
-
-        # return song submission sorted from energy least to greatest
-
-        # complete this
-
-        # sorts by energy
-        return song_submissions_scored
-
-    @staticmethod
-    def get_peak_sort(song_submissions_scored: Iterable["SongSubmission"]):
-
-        # song_submissions_scored is a list of song submissions
-
-        # that have been scored with current comparisons
-
-        # sorts by peakyness
-        return song_submissions_scored
-
-    @staticmethod
-    def mvg_avg(song_submissions_sorted: Iterable["SongSubmission"]):
 
+    def mvg_avg(self,song_submissions_sorted: Iterable["SongSubmission"]):
         mvg_avg = [None] * len(song_submissions_sorted)
 
         # song_submissions_scored is a list of song submissions
-
         # that have been sorted into prepeak,peak,and post peak
-
         # these prepeak, peak, and post posteak have each been sorted
         # accordingly with energy
-
         # that have been scored with current comparisons
-
         # get a moving average corresponding to quality score
 
         return mvg_avg
 
-    @staticmethod
-    def check_quality(
+    def check_quality(self,
             song_submissions_sorted: Iterable["SongSubmission"],
             remove: int):
 
         song_submissions_pruned = song_submissions_sorted
 
         # do (song_submissions_sorted[i].quality_score - mvg_avg[i])
-
         # for every song in score submission
-
         # return n indices with smallest values
-
         # these indices are to be removed
+
         return song_submissions_pruned
 
-    @staticmethod
-    def get_final_list(comparison_submissions: Iterable["SongComparison"],
+    def get_dict_from_keys(self, new_keys):
+        new_dict=dict()
+        new_dict={key: self.song_score_dict[key] for key in new_keys}
+        return new_dict
+    
+    def get_final_list(self):
```

Comment:
This advice could apply in a lot of places, but can we please add return type annotations to ad many functions as possible, especially those that are longish and don't end with something with obvious type like a literal or comprehension?

Reply:
fair point these will be added